extensions/binding/org.eclipse.smarthome.binding.ble/src/main/java/org/eclipse/smarthome/binding/ble/internal/BleHandlerFactory.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 1997, 2015 by Huawei Technologies Co., Ltd. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.ble.internal;
+
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.eclipse.smarthome.binding.ble.BleBindingConstants;
+import org.eclipse.smarthome.binding.ble.handler.BleBaseThingHandler;
+import org.eclipse.smarthome.binding.ble.handler.ParrotFlowerPowerThingHandler;
+import org.eclipse.smarthome.binding.ble.handler.WiTEnergyThingHandler;
+import org.eclipse.smarthome.binding.ble.handler.YeelightBlueThingHandler;
+import org.eclipse.smarthome.binding.ble.internal.discovery.BleDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BleHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Chris Jackson - Initial Contribution
+ */
+public class BleHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(BleHandlerFactory.class);
+
+    Map<ThingTypeUID, Class<? extends BaseThingHandler>> thingHandlers = new HashMap<ThingTypeUID, Class<? extends BaseThingHandler>>();
+
+    private BleDiscoveryService adapterDiscoveryService;
+    private ServiceRegistration<?> serviceReg;
+
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+
+        thingHandlers.put(BleBindingConstants.THING_TYPE_GENERIC, BleBaseThingHandler.class);
```

Comment:
I think this line should be `thingHandlers.put(BleBindingConstants.THING_TYPE_GENERIC, BleGenericHandler.class);` as BleBaseThingHandler is abstract.